src/trio/_core/_run.py

```diff
@@ -1404,10 +1402,9 @@ def __del__(self) -> None:
 
 @final
 @attrs.define(eq=False, repr=False)
-class Task(metaclass=NoPublicConstructor):  # type: ignore[misc]
+class Task(metaclass=NoPublicConstructor):  # type: ignore[explicit-any]
     _parent_nursery: Nursery | None
-    # Explicit "Any" is not allowed
-    coro: Coroutine[Any, Outcome[object], Any]  # type: ignore[misc]
+    coro: types.CoroutineType[Any, Outcome[object], Any]  # type: ignore[explicit-any]
```

Comment:
Curious why this needs to be `types.CoroutineType` and not just `collections.abc.Coroutine`

Reply:
Because asyncio.iscoroutine is a type guard for Coroutine and it can pass things that aren't CoroutineType. At least on Python <3.11 and probably later.

(I mean more generally the Coroutine abc doesn't require the cr_frame etc attributes but that's a circular argument.)